src/Controllers/ElementalAreaController.php

```diff
@@ -195,25 +228,123 @@ public function formAction(HTTPRequest $request)
-     *
-     * @param array $data
-     * @param int $elementID
-     * @return array
+     * Create a form for a given element
      */
-    public static function removeNamespacesFromFields(array $data, $elementID)
+    private function createElementForm(BaseElement $element): Form
     {
-        $output = [];
-        $template = sprintf(EditFormFactory::FIELD_NAMESPACE_TEMPLATE ?? '', $elementID, '');
-        foreach ($data as $key => $value) {
-            // Only look at fields that match the namespace template
-            if (substr($key ?? '', 0, strlen($template ?? '')) !== $template) {
+        $id = $element->ID;
+        $scaffolder = Injector::inst()->get(EditFormFactory::class);
+
+        /** @var Form $form */
+        $form = $scaffolder->getForm(
+            $this,
+            sprintf(static::FORM_NAME_TEMPLATE, $id),
+            ['Record' => $element]
+        );
+
+        $urlSegment = $this->config()->get('url_segment');
+        $form->setFormAction("admin/$urlSegment/elementForm/$id");
+
+        if (!$element->canEdit()) {
+            $form->makeReadonly();
+        }
+
+        $form->addExtraClass('element-editor-editform__form');
+        $form->addExtraClass('bypass-entwine-submission');
+
+        // Add a save button to the form
+        // This button needs to be present for the save() method to be called
+        // It will be hidden with CSS via the btn--hidden class
+        $form->setActions(FieldList::create([
+            FormAction::create('save', 'Save')
+                ->addExtraClass('btn--hidden')
+        ]));
+
+        // Set the form request handler to return a FormSchema response during a POST request
+        // This will override the default FormRequestHandler::getAjaxErrorResponse() which isn't useful
+        $form->setValidationResponseCallback(function (ValidationResult $errors) use ($form, $id) {
+            $schemaId = Controller::join_links(
+                $this->Link('schema'),
+                $this->config()->get('url_segment'),
+                $id
+            );
+            return $this->getSchemaResponse($schemaId, $form, $errors);
+        });
+
+        return $form;
+    }
+
+    /**
+     * Call model validation taking into account the namespaced fields
+     */
+    private function getElementValidationResult(
+        Form $form,
+        array $data,
+        BaseElement $element
+    ): ValidationResult {
+
+        // Remove the pseudo namespaces that were added by EditFormFactory
+        $dataWithoutNamespaces = $this->removeNamespacesFromFields($data, $element->ID);
+
+        // Create a seperate form because we're going to modify it
+        // if we simply updated $form it will mess up the FormSchema response, even if we `clone $form`
+        $formWithoutNamespaces = $this->createElementForm($element);
+
+        // Remove pseudo namespaces from fields so that something like RequiredFields('Title') works
```

Comment:
Have updated to 'namespace prefixes'

Have updated comment to simply be `Remove namespace prefixes from fields so that validation works` - it's needed for model level validation.

`save()` seems like a bad place to be worrying about removing namespaces, currently it's doing checks that return HTTP status codes and calling this validation method and write(), it's all kind of high level.  Seems far better to have the namespace stuff neatly tucked away in this private method

Reply:
> `save()` seems like a bad place to be worrying about removing namespaces, currently it's doing checks that return HTTP status codes and calling this validation method and write(), it's all kind of high level. Seems far better to have the namespace stuff neatly tucked away in this private method

Sorry, the comment about `save()` was meant to be about `getElementValidationResult()`, not about `removeNamespacesFromFields()`.

In the `save()` method, we call `write()`. This results in calling `validate()` and throws a `ValidationException` if there are errors.

Seems to me we should just catch the `ValidationException` at **that** point, handle any namespace changes that are needed in the validation result, and throw the `ValidationException` again from there.

Reply:
Updated

I've also just removed some unused code from that method that looked like it was originally doing Form validation, which is unnecessary. I think with was copied from the original spike